plugins/zoom-to-fit/src/index.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview UI control for zoom-to-fit.
+ */
+
+import Blockly from 'blockly/core';
+
+/**
+ * Class for zoom to fit control.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+ * @implements {Blockly.IPositionable}
+ * @constructor
+ */
+export class ZoomToFitControl {
+  /**
+   * Constructor for zoom-to-fit control.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the plugin will
+   *     be added to.
+   */
+  constructor(workspace) {
+    /**
+     * The workspace.
+     * @type {!Blockly.WorkspaceSvg}
+     * @protected
+     */
+    this.workspace_ = workspace;
+
+    /**
+     * The SVG group containing the zoom controls.
+     * @type {SVGElement}
+     * @private
+     */
+    this.svgGroup_ = null;
+
+    /**
+     * Left coordinate of the zoom controls.
+     * @type {number}
+     * @private
+     */
+    this.left_ = 0;
+
+    /**
+     * Top coordinate of the zoom controls.
+     * @type {number}
+     * @private
+     */
+    this.top_ = 0;
+
+    /**
+     * Width of the zoom controls.
+     * @type {number}
+     * @private
+     */
+    this.WIDTH_ = 32;
+
+    /**
+     * Height of the zoom controls.
+     * @type {number}
+     * @private
+     */
+    this.HEIGHT_ = 32;
+
+    /**
+     * Distance between zoom controls and bottom edge of workspace.
+     * @type {number}
+     * @private
+     */
+    this.MARGIN_BOTTOM_ = 20;
+
+    /**
+     * Distance between zoom controls and right edge of workspace.
+     * @type {number}
+     * @private
+     */
+    this.MARGIN_SIDE_ = 20;
+
+    /**
+     * The starting vertical distance between the workspace edge and the
+     * control. The value is initialized during `init`.
+     * @type {?number}
+     * @private
+     */
+    this.verticalSpacing_ = null;
```

Comment:
This can be 0 until initialized. Then the types are simplified, since it's always valid/never null.